src/api/spec/cassettes/Webui_PackageController/POST_branch/with_currrent_revision_parameter/1_4_8_3.yml

```diff
@@ -0,0 +1,697 @@
+    body:
+      encoding: UTF-8
+      string: |
+        <package name="package_with_revisions" project="home:tom">
+          <title>If I Forget Thee Jerusalem</title>
+          <description>Dolorem cum optio beatae.</description>
+        </package>
+    headers:
+      Accept-Encoding:
+      - gzip;q=1.0,deflate;q=0.6,identity;q=0.3
+      Accept:
+      - "*/*"
+      User-Agent:
+      - Ruby
+  response:
+    status:
+      code: 200
+      message: OK
+    headers:
+      Content-Type:
+      - text/xml
+      Cache-Control:
+      - no-cache
+      Connection:
+      - close
+      Content-Length:
+      - '169'
+    body:
+      encoding: UTF-8
+      string: |
+        <package name="package_with_revisions" project="home:tom">
+          <title>If I Forget Thee Jerusalem</title>
+          <description>Dolorem cum optio beatae.</description>
+        </package>
+    http_version: 
+  recorded_at: Tue, 11 Sep 2018 15:56:06 GMT
+- request:
+    method: put
+    uri: http://backend:5352/source/home:tom/package_with_revisions/somefile.txt
+    body:
+      encoding: US-ASCII
+      string: '0'
+    headers:
+      Accept-Encoding:
+      - gzip;q=1.0,deflate;q=0.6,identity;q=0.3
+      Accept:
+      - "*/*"
+      User-Agent:
+      - Ruby
+  response:
+    status:
+      code: 200
+      message: OK
+    headers:
+      Content-Type:
+      - text/xml
+      Cache-Control:
+      - no-cache
+      Connection:
+      - close
+      Content-Length:
+      - '211'
+    body:
+      encoding: UTF-8
+      string: |
+        <revision rev="119" vrev="119">
+          <srcmd5>cabf9a1f9a0b7c19d4f7ff46166f4ee0</srcmd5>
+          <version>unknown</version>
+          <time>1536681366</time>
+          <user>unknown</user>
+          <comment></comment>
+          <requestid/>
+        </revision>
+    http_version: 
+  recorded_at: Tue, 11 Sep 2018 15:56:06 GMT
+- request:
+    method: put
+    uri: http://backend:5352/source/home:tom/package_with_revisions/somefile.txt
+    body:
+      encoding: US-ASCII
+      string: '1'
+    headers:
+      Accept-Encoding:
+      - gzip;q=1.0,deflate;q=0.6,identity;q=0.3
+      Accept:
+      - "*/*"
+      User-Agent:
+      - Ruby
+  response:
+    status:
+      code: 200
+      message: OK
+    headers:
+      Content-Type:
+      - text/xml
+      Cache-Control:
+      - no-cache
+      Connection:
+      - close
+      Content-Length:
+      - '211'
+    body:
+      encoding: UTF-8
+      string: |
+        <revision rev="120" vrev="120">
+          <srcmd5>efbe5f0a5dd48df5129b4319df43aa45</srcmd5>
+          <version>unknown</version>
+          <time>1536681366</time>
+          <user>unknown</user>
+          <comment></comment>
+          <requestid/>
+        </revision>
+    http_version: 
+  recorded_at: Tue, 11 Sep 2018 15:56:06 GMT
+- request:
+    method: get
+    uri: http://backend:5352/source/home:tom/package_with_revisions?rev=2
```

Comment:
I find this quite dangerous. Because as your cassette shows your package_with_revisions only creates revision 119 - and then asks for revision 2. Of course the running backend has a revision 2 - but as your test didn't create it, it could just as well be broken, and your test will fail.

It would be better if you either use a FactoryBot generate package name, so you can be certain your revision 2 is really your revision 2. Or you mock the fetching of rev=2 to avoid cassette usage for it.

Reply:
@coolo, but the factory `package_with_revisions` create by default 2 revisions.

Reply:
right, it creates *two more* - that doesn't mean rev=2 is the one your factory just created. As this one is created in the backend.

So what you would need to do to make sure it's rev=2 also in the backend is deleting it on the backend first